037-pluggable-pod-security-profiles.md

```diff
@@ -0,0 +1,234 @@
+
+## Current situation
+
+Currently, Strimzi provides a minimal configuration of the security context by default.
+When running on Kubernetes and using persistent storage, we configure the group which should be used for the storage.
+That is the only automatic configuration Strimzi makes.
+
+Users can use the `.template` properties in the Strimzi custom resources to configure their own security context.
+This allows them to customize a security context that matches their own requirements and policies.
+On OpenShift, the security context is automatically injected into the Pods when they are created based on OpenShift Security Context Constraints (SCC) policies.
+This typically involves dropping some capabilities, using some random unprivileged user ID etc.
+In some cases, Strimzi users might also use similar systems to inject the security context using admission controllers and similar tools.
+
+The _baseline_ profile basically means that you just use the default settings without requesting any additional privileges or capabilities.
+So without any of the additional tooling mentioned previously, Strimzi operators and the Pods created by them at present match the _baseline_ profile.
+If you try to run Strimzi in a cluster where the _restricted_ profile is enforced, it does not work out of the box.
+
+Apart from the Kaniko builder used for the Kafka Connect build, all our components are able to run under the _restricted_ profile.
+But the operator doesn't configure the appropriate security context and the admission plugin will reject them.
+Users have to _manually_ add the matching security context configuration into the `.template` sections of the custom resources and into the operator deployments to allow the pods to be created under the _restricted_ profile.
+To match the _restricted_ profile, users must configure the following security context:
+
+```yaml
+          securityContext:
+            allowPrivilegeEscalation: false
+            capabilities:
+              # The restricted allows the use of the NET_BIND_SERVICE capability.
+              # But Strimzi does not require it for anything, so we do not include it here.
+              drop:
+                - ALL
+            runAsNonRoot: true
+            seccompProfile:
+              type: RuntimeDefault
+```
+
+While users can configure the security context, it is not user-friendly.
+The Pod security context does not cover all the different options required by the _restricted_ profile.
+So the security context has to be configured for every container (including init containers, sidecars etc.).
+So a single Kafka cluster might need configuration in 9 different places in the `Kafka` custom resource: ZooKeeper, Kafka broker, Kafka init-container, Topic Operator, User Operator, TLS sidecar, Kafka Exporter, Cruise Control, and another TLS sidecar.
```

Comment:
"another TLS sidecar" ? Referring to the CC one, it was removed. Anyway it's not so important here.